### ISSUE TITLE
Make undeploy takes a long time to complete

### ISSUE THREAD

**Issue by maryamtahhan** [bpfman/bpfman#915]
`make undeploy` seems to take a very long time to complete. If it's interrupted or forced, then it's important for the end user to unmount a number of dirs in /run/bpfd and delete it manually

```
if [ -d /run/bpfd ]; then
umount /run/bpfd/fs; umount /run/bpfd/csi;  rm -rf /run/bpfd/
fi

if [ -d /run/bpfman ]; then
umount /run/bpfman/fs; umount /run/bpfman/csi;  umount /run/bpfman/sock; rm -rf /run/bpfman/
fi
```

else they will encounter an issue on the next attempt to `make deploy`:

```
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  42s                default-scheduler  Successfully assigned bpfd/bpfd-daemon-4c922 to wsfd-advnetlab47.anl.lab.eng.bos.redhat.com
  Normal   Pulling    41s                kubelet            Pulling image "quay.io/fedora/fedora-minimal:latest"
  Normal   Pulled     41s                kubelet            Successfully pulled image "quay.io/fedora/fedora-minimal:latest" in 269ms (269ms including waiting)
  Normal   Created    41s                kubelet            Created container mount-bpffs
  Normal   Started    41s                kubelet            Started container mount-bpffs
  Normal   Created    40s                kubelet            Created container node-driver-registrar
  Normal   Pulled     40s                kubelet            Container image "quay.io/bpfd/csi-node-driver-registrar:v2.9.0" already present on machine
  Normal   Started    40s                kubelet            Started container node-driver-registrar
  Warning  Failed     39s (x2 over 40s)  kubelet            Error: failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/var/lib/kubelet/plugins/csi-bpfd" to rootfs at "/run/bpfd/csi": possibly malicious path detected -- refusing to operate on /run/containerd/io.containerd.runtime.v2.task/k8s.io/bpfd/rootfs/run/bpfd/csi (deleted): unknown
  Normal   Pulled     39s (x2 over 40s)  kubelet            Container image "quay.io/bpfd/bpfd-agent:latest" already present on machine
  Normal   Created    39s (x2 over 40s)  kubelet            Created container bpfd-agent
  Normal   Started    39s (x2 over 40s)  kubelet            Started container bpfd-agent
  Warning  BackOff    37s (x2 over 38s)  kubelet            Back-off restarting failed container bpfd in pod bpfd-daemon-4c922_bpfd(4030b300-8df6-4f8b-9d40-c1058419eb40)
  Warning  BackOff    37s (x2 over 38s)  kubelet            Back-off restarting failed container bpfd-agent in pod bpfd-daemon-4c922_bpfd(4030b300-8df6-4f8b-9d40-c1058419eb40)
  Normal   Created    24s (x3 over 40s)  kubelet            Created container bpfd
  Normal   Pulled     24s (x3 over 40s)  kubelet            Container image "quay.io/bpfd/bpfd:latest" already present on machine
```

**Comment by astoycos** [comment]
@maryamtahhan Thanks for opening this!! After looking`make undeploy` needs to be updated slightly to actually clean everything up I think.... The flow should look like this 

1. Delete the bpfman configmap, which will trigger the operator to remove the bpfman damonset `kubectl delete -n bpfman cm bpfman-config`
2. Delete the namespace to cleanup the operator and all remaining resources


If 1. isn't completed before 2. The bpfman-ds deletion could be hung with a finalizer that the operator was supposed to remove before exiting... I _think_ this should fix things!

**Comment by astoycos** [comment]
Partially fixed by #916 But I'm still getting 

```
message: 'failed to create containerd task: failed to create shim task: OCI
          runtime create failed: runc create failed: unable to start container process:
          error during container init: error mounting "/var/lib/kubelet/pods/daa0bfd7-804a-4374-9dad-668223070372/volumes/kubernetes.io~empty-dir/bpfman-sock"
          to rootfs at "/run/bpfman/sock": possibly malicious path detected -- refusing
          to operate on /run/containerd/io.containerd.runtime.v2.task/k8s.io/b2742d0430c69e74cba6ea155a7008517c832a70e9d92da21598036337ae0bb2/rootfs/run/bpfman/sock
```

On the re-deploy

**Comment by astoycos** [comment]
Fixed in #962 and #964
